discovery-provider/src/api/v1/tracks.py

```diff
@@ -308,6 +308,17 @@ def get(self, track_id):
         if not creator_nodes:
             abort_not_found(track_id, ns)
 
+        # before redirecting to content node,
+        # make sure the track isn't deleted and the user isn't deactivated
+        args = {
+            "id": [decoded_id],
+            "with_users": True,
+        }
+        tracks = get_tracks(args)
```

Comment:
i think the extra latency here would be fine. i like this solution (just musing aloud)